packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2742,6 +2747,10 @@ class InputDecoration {
          !(label != null && labelText != null),
          'Declaring both label and labelText is not supported.',
        ),
+       assert(
+         !(hintText != null && hint != null),
```

Comment:
Nit: Another option is `hintText == null || hint == null`, either way is fine though.

Reply:
updated